### PR TITLE
fix: Correctly set notifier name/version for iOS native crashes

### PR DESCRIPTION
**Pull request by kattrali**
Tested by booting an example app and inspecting notifier info in the payload and on the dashboard.

### CODE REVIEW

File: src/BugsnagUnity.mm [hunk 1]
```diff
@@ -135,8 +139,17 @@ void bugsnag_setMetadata(const void *configuration, const char *tab, const char
...
 
-void bugsnag_startBugsnagWithConfiguration(const void *configuration) {
+void bugsnag_startBugsnagWithConfiguration(const void *configuration, char *notifierVersion) {
   [Bugsnag startBugsnagWithConfiguration: (__bridge BugsnagConfiguration *)configuration];
+  if (notifierVersion != NULL) {
+    NSString *ns_version = [NSString stringWithUTF8String:notifierVersion];
+    id notifier = [Bugsnag notifier];
+    [notifier setValue:@{
+        @"version": ns_version,
+        @"name": @"Unity Bugsnag Notifier",
```
Comment (snmaynard) [inline src/BugsnagUnity.mm]: Should be `Bugsnag Unity (Objective-C)`? cc @duncanhewett @bullmo

Comment (kattrali) [reply src/BugsnagUnity.mm]: It was `Bugsnag Unity (Cocoa)` before. I made it match what android does now, which makes more sense?

Comment (kattrali) [reply src/BugsnagUnity.mm]: If it needs to be `Bugsnag Unity (Cocoa)`, I'll update android to match too.

Comment (snmaynard) [reply src/BugsnagUnity.mm]: In the spec it says `Bugsnag <Framework> (Dependency)`. The cocoa notifier seems to be `objective-c` -> https://github.com/bugsnag/bugsnag-cocoa/blob/master/Source/BugsnagNotifier.m#L216

Comment (snmaynard) [reply src/BugsnagUnity.mm]: The pipeline needs to know about any notifier name changes ofc @mikeewhite too

Comment (kattrali) [reply src/BugsnagUnity.mm]: The cocoa notifier gets overwritten as `Bugsnag <Platform> Notifier`: 

https://github.com/bugsnag/bugsnag-cocoa/blob/master/Source/BugsnagNotifier.m#L336-L341

Comment (kattrali) [reply src/BugsnagUnity.mm]: To clarify this change:

* In the latest released version of bugsnag-unity, it reports the cocoa notifier info for native crashes, which right now is `iOS Bugsnag Notifier` v5.18.0. On Android it uses `Unity Bugsnag Notifier` in all cases.
* In bugsnag-unity v3.x, it reported the name as `Bugsnag Unity (Cocoa)` or `Bugsnag Unity (Android)` and the unity notifier version number.

Comment (snmaynard) [reply src/BugsnagUnity.mm]: Lets do the sensible thing™ when comparing the spec, and what the various notifiers and pipeline currently do. Your call @kattrali.

Comment (kattrali) [reply src/BugsnagUnity.mm]: I have consulted with The Oracle (@bullmo) and will make a slight adjustment.

**Pull request merged**